src/ert/_c_wrappers/enkf/site_config.py

```diff
@@ -1,59 +1,119 @@
-
-        super().__init__(c_ptr)
+from ert._c_wrappers.job_queue import EnvironmentVarlist, ExtJob, ExtJoblist
+
+
+@dataclass
+class SiteConfig:
+    job_list: ExtJoblist
+    var_list: EnvironmentVarlist
+
+    @classmethod
+    def _add_config_content(
+        cls,
+        license_root_path: str,
+        site_config_content: ConfigContent,
+        job_list: ExtJoblist,
+        var_list: EnvironmentVarlist,
+    ):
+
+        if site_config_content.hasKey(ConfigKeys.INSTALL_JOB):
+            for args in iter(site_config_content[ConfigKeys.INSTALL_JOB]):
+                if not os.path.isfile(args[1]):
+                    print(f"WARNING: Unable to locate job file {args}")
+                    continue
+                try:
+                    new_job = ExtJob(
+                        config_file=args[1],
+                        private=False,
+                        name=args[0],
+                        license_root_path=license_root_path,
+                    )
+                    new_job.convertToCReference(None)
+                    job_list.add_job(new_job.name(), new_job)
+                except (ValueError, OSError):
+                    print(f"WARNING: Unable to create job from {args}")
+
+        if site_config_content.hasKey(ConfigKeys.INSTALL_JOB_DIRECTORY):
+            for args in iter(site_config_content[ConfigKeys.INSTALL_JOB_DIRECTORY]):
+                job_path = args[0]
+                if not os.path.isdir(job_path):
+                    print(f"WARNING: Unable to locate job directory {job_path}")
+                    continue
+                files = os.listdir(job_path)
+                for file_name in files:
+                    full_path = os.path.join(job_path, file_name)
+                    if os.path.isfile(full_path):
+                        try:
+                            new_job = ExtJob(
+                                config_file=full_path,
+                                private=False,
+                                license_root_path=license_root_path,
+                            )
+                            new_job.convertToCReference(None)
+                            job_list.add_job(new_job.name(), new_job)
+                        except (ValueError, OSError):
+                            print(
+                                f"WARNING: Unable to create job from {full_path} "
+                                f"in directory {job_path}"
+                            )
+
+        site_environment_vars = site_config_content[ConfigKeys.SETENV]
+
+        for elem in iter(site_environment_vars):
+            var_list.setenv(elem[0], elem[1])
+
+        site_paths = site_config_content["UPDATE_PATH"]
+
+        for elem in iter(site_paths):
+            var_list.update_path(elem[0], elem[1])
+
+    @classmethod
+    def from_config_content(
+        cls, config_content: ConfigContent, site_config_content: ConfigContent
```

Comment:
maybe you could rename `config_content` into `user_config_content`? to follow the newly introduced duality of user vs site config?

Reply:
Good idea.
It's not newly introduced though. Just newly made visible for sane people.

Reply:
I see : D